src/buffer_manager.ts

```diff
@@ -808,3 +820,51 @@ export class BufferManager implements Disposable, NeovimRedrawProcessable, Neovi
         }
     }
 }
+
+type BufferChangeCallback = BufferManager["receivedBufferEvent"];
+
+/**
+ * Implements the VSCode document provider API for external buffers from neovim.
+ */
+class ExternalBufferProvider implements TextDocumentContentProvider {
```

Comment:
nit: "External" is over-explaining isn't it? Is there any other kind of "buffer provider"? 

```suggestion
class BufferProvider implements TextDocumentContentProvider {
```

Reply:
Not currently, but there could be other `CustomTextEditorProvider`s or similar for some of the other use cases mentioned. Since this provider is only used for "external buffers" coming directly from Neovim I thought it might be helpful to name it as such.

If you think it's overly repetitive I can definitely rename though.

Reply:
> but there could be other `CustomTextEditorProvider`s or similar for some of the other use cases mentioned.

When that happens, then verbose names can be used. But until then there's no need for a verbose name.